### PR TITLE
ZIOS-11401: Allow users to accept legal hold requests

### PR DESCRIPTION
**Pull request by alexaubry**
### Issues

We want users to be notified upon reception of the legal hold request event, and be able to enter their password to confirm the request.

### Solutions

- Observe the event on the client view controller to present the alert
- Observe the event on the conversation list top bar to show the pending request button
- Add more robust logic to handle user input from the alert

### Dependencies

Needs releases with:

- [x] https://github.com/wireapp/wire-ios-sync-engine/pull/1036

### CODE REVIEW

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -6917,7 +6929,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "# Run the update_stylekit package\nswift run --package-path Scripts/update_stylekit/\n";
+			shellScript = "# Run the update_stylekit package\n# swift run --package-path Scripts/update_stylekit/\n";
```
Comment (typfel) [inline Wire-iOS.xcodeproj/project.pbxproj]: Should this be commented out?

File: Wire-iOS/Sources/UserInterface/Helpers/UserInputRequest+LegalHold.swift [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+
+extension UserType where Self: SelfLegalHoldSubject {
+
+    /**
+     * Creates the password input request to respond to a legal hold activation requesr from the team admin.
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Helpers/UserInputRequest+LegalHold.swift]:
```suggestion
     * Creates the password input request to respond to a legal hold activation request from the team admin.
```

File: Wire-iOS.xcodeproj/project.pbxproj (hunk 1 above)
Comment (alexaubry) [reply Wire-iOS.xcodeproj/project.pbxproj]: Good catch!

**Comment by typfel** [comment]
Can we do `Display alert when LH is accepted on another device` in another PR? it would be nice to get this merged

**Pull request merged**